homeassistant/components/minecraft_server/sensor.py

```diff
@@ -54,92 +121,44 @@ async def async_setup_entry(
 class MinecraftServerSensorEntity(MinecraftServerEntity, SensorEntity):
     """Representation of a Minecraft Server sensor base entity."""
 
+    entity_description: MinecraftServerSensorEntityDescription
+
     def __init__(
         self,
         server: MinecraftServer,
-        entity_type: str,
-        icon: str,
-        unit: str | None = None,
-        device_class: str | None = None,
+        description: MinecraftServerSensorEntityDescription,
     ) -> None:
         """Initialize sensor base entity."""
-        super().__init__(server, entity_type, icon, device_class)
-        self._attr_native_unit_of_measurement = unit
+        super().__init__(server)
+        self.entity_description = description
+        self._attr_unique_id = f"{server.unique_id}-{self.entity_description.key}"
 
     @property
     def available(self) -> bool:
         """Return sensor availability."""
         return self._server.online
 
-
-class MinecraftServerVersionSensor(MinecraftServerSensorEntity):
-    """Representation of a Minecraft Server version sensor."""
-
-    _attr_translation_key = KEY_VERSION
-
-    def __init__(self, server: MinecraftServer) -> None:
-        """Initialize version sensor."""
-        super().__init__(server=server, entity_type=KEY_VERSION, icon=ICON_VERSION)
-
-    async def async_update(self) -> None:
-        """Update version."""
-        self._attr_native_value = self._server.data.version
-
-
-class MinecraftServerProtocolVersionSensor(MinecraftServerSensorEntity):
-    """Representation of a Minecraft Server protocol version sensor."""
-
-    _attr_translation_key = KEY_PROTOCOL_VERSION
-
-    def __init__(self, server: MinecraftServer) -> None:
-        """Initialize protocol version sensor."""
-        super().__init__(
-            server=server,
-            entity_type=KEY_PROTOCOL_VERSION,
-            icon=ICON_PROTOCOL_VERSION,
-        )
-
-    async def async_update(self) -> None:
-        """Update protocol version."""
-        self._attr_native_value = self._server.data.protocol_version
-
-
-class MinecraftServerLatencySensor(MinecraftServerSensorEntity):
-    """Representation of a Minecraft Server latency sensor."""
-
-    _attr_translation_key = KEY_LATENCY
-
-    def __init__(self, server: MinecraftServer) -> None:
-        """Initialize latency sensor."""
-        super().__init__(
-            server=server,
-            entity_type=KEY_LATENCY,
-            icon=ICON_LATENCY,
-            unit=UnitOfTime.MILLISECONDS,
-        )
-
     async def async_update(self) -> None:
-        """Update latency."""
-        self._attr_native_value = self._server.data.latency
+        """Update sensor state."""
+        self._attr_native_value = self.entity_description.value_fn(self._server.data)
 
 
 class MinecraftServerPlayersOnlineSensor(MinecraftServerSensorEntity):
```

Comment:
I think you still split this sensor off because of the state attributes, but you can use a function for that. Checkout the youtube sensors, one of the sensors is for subscribers, while the other is for the latest video. the latest video has the video id in the attributes while the other one doesnt

Reply:
I see, good idea. I'll try this out 👍